### PR TITLE
HEYK-187: Disconnect timeout

### PR DESCRIPTION
**Pull request by ekorolev**

### CODE REVIEW

File: src/api/socket/client.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ function connect() {
     }
 
     client.once('connect', data => {
+      client.lastSocketId = client.id;
+      console.log('ZHSBS client', client);
```
Comment (ekorolev) [inline src/api/socket/client.js]:
```suggestion
```

File: src/api/socket/index.js [hunk 2]
```diff
@@ -53,36 +67,44 @@ export async function destroy() {
...
+    const authData = {
       transaction: 'auth',
       workspaceId: store.getters['me/getSelectedWorkspaceId'],
       token: accessToken,
       onlineStatus: onlineStatus,
-    });
+      ...(prevSocketId ? { prevSocketId } : prevSocketId),
+    };
+
+    console.log('Auth with data', authData);
```
Comment (ekorolev) [inline src/api/socket/index.js]:
```suggestion
```

File: src/renderer/store/actions.js [hunk 3]
```diff
@@ -125,11 +144,82 @@ export default {
...
+      commit('SET_SOCKET_CONNECTED', false);
+    } else {
+      // Handle socket connected
+      commit('SET_SOCKET_CONNECTED', true);
+
+      if (authData.reconnected) {
+        // if server said that channel is selected
+        // then select channel in app
+        if (authData.channelId && getters['me/getSelectedChannelId'] !== authData.channelId) {
+          commit('janus/SET_OPTIONS', authData);
```
Comment (ekorolev) [inline src/renderer/store/actions.js]: selectChannel without api call

File: src/renderer/store/actions.js [hunk 4]
```diff
@@ -125,11 +144,82 @@ export default {
...
+
+          if (state.me.mediaState.microphone === true) {
+            ipcRenderer.send('tray-animation', true);
+          }
+        }
+
+        // if server said that channel not selected
+        // unselect channel in app
+        if (!authData.channelId && getters['me/getSelectedChannelId']) {
+          commit('channels/REMOVE_USER', {
```
Comment (ekorolev) [inline src/renderer/store/actions.js]: unselectChannel without api call

File: src/renderer/store/actions.js [hunk 5]
```diff
@@ -125,11 +144,82 @@ export default {
...
+          dispatch('me/setDefaultMediaState');
+
+          callWindow.hideAll();
+
+          ipcRenderer.send('tray-animation', false);
+        }
+      }
+
+      if (!authData.reconnected && getters['me/getSelectedChannelId']) {
+        commit('channels/REMOVE_USER', {
```
Comment (ekorolev) [inline src/renderer/store/actions.js]: unselectCHANNEL WITHOUT API

**Review by xinger — APPROVED** [review]

**Pull request merged**